pkg/build/builder/common.go

```diff
@@ -425,19 +440,23 @@ func addBuildParameters(dir string, build *buildapiv1.Build, sourceInfo *git.Sou
 		return err
 	}
 
-	if err := replaceImagesFromSource(node, build.Spec.Source.Images); err != nil {
+	var buildArgs []corev1.EnvVar
```

Comment:
Looks like you discovered a separate issue from the `ARG` one where docker strategy ENVs were getting lost as well @nalind 

I'm I reading that correctly?

If so, would a comment, analogous to your ARG explanation above, placed here, and a tweak to the PR title/commit, seem prudent?

Reply:
And some new unit tests as well, again assuming I'm reading this correctly.

Reply:
They're still build args.  The API struct is just called an `EnvVar`, I'm guessing because inventing a new type for every key-value pairing would have been excessive.

I'm not sure this added logic buys us anything, since we don't do any substitutions when we're walking the parsed tree, but I added it for completeness.  I wouldn't object if we wanted to drop it.

Reply:
Thanks for the clarification.

I don't feel either way about keeping it at the moment, though perhaps a unit test would prove one way or the other whether it buys us anything.

Unless other chime in with stronger opinions, I'll leave it at that.